core/modules/wmgr/python/dbMgr.py

```diff
@@ -589,14 +598,40 @@ def deleteChunk(dbName, tblName, chunkId):
 @dbService.route('/<dbName>/tables/<tblName>/data', methods=['POST'])
 @dbService.route('/<dbName>/tables/<tblName>/chunks/<int:chunkId>/data', methods=['POST'])
 @dbService.route('/<dbName>/tables/<tblName>/chunks/<int:chunkId>/overlap', methods=['POST'])
-def locadData(dbName, tblName, chunkId=None):
+def loadData(dbName, tblName, chunkId=None):
     """
     Upload data into a table or chunk using the file format supported by mysql
     command LOAD DATA [LOCAL] INFILE.
 
-    Not implemented yet.
+    For this method we expect all data to come in one request, and in addition
+    to table data we also need few parameters which define how data is formatted
+    (things like column separator, line terminator, escape, etc.) The whole
+    request must be multipart/form-data and contain two parts:
+    - one with the name "load-options" which contains set of options encoded
+      with usual application/x-www-form-urlencoded content type, options are:
+      - delimiter - defaults to TAB
+      - enclose - defaults to double quote
+      - escape - defaults to backslash
+      - terminate - defaults to newline
+      - compressed - "0" or "1", by default is guessed from file extension (.gz)
+    - one with the file data (name "table-data"), the data come in original
+      format with binary/octet-stream content type and binary encoding, and it
+      may be compressed with gzip.
     """
 
+    def _copy(tableData, fifoName, compressed):
+        """ Helper method to run data copy in a separate thread """
+        if compressed:
+            src = gzip.GzipFile(fileobj=tableData.stream, mode='rb')
+            dst = open(fifoName, 'wb')
+            shutil.copyfileobj(src, dst)
+            _log.info('uncompressed table data to file %s', fifoName)
+            dst.close()
+            src.close()
```

Comment:
Would it be a bit safer to use context managers for src and dst here (in conjunction with contextlib.closing if necessary)?